### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by delcin-raj**

### CODE REVIEW

**Comment by delcin-raj** [comment]
Not sure how `Cargo.lock` got committed.
It was listed in .gitignore though.

File: src/algorithms/bnb.rs [hunk 1]
```diff
@@ -1,13 +1,22 @@
...
-        WasteMetric,
-    },
+    types::{CoinSelectionOpt, OutputGroup, SelectionError, SelectionOutput, WasteMetric},
     utils::{calculate_fee, calculate_waste, effective_value},
 };
 
+/// Struct for three arguments : target_for_match, match_range and target_feerate
+///
+/// Wrapped in a struct or else input for fn bnb takes too many arguments - 9/7
+/// Leading to usage of stack instead of registers - https://users.rust-lang.org/t/avoiding-too-many-arguments-passing-to-a-function/103581
```
Comment (NeoZ666) [inline src/algorithms/bnb.rs:11]: This was added by me, but sounds a bit unprofessional, plus a link in the comments. Should it be removed?

Comment (delcin-raj) [reply src/algorithms/bnb.rs:11]: I noticed it, can you come up with good comment?
It's a private struct we can just remove the comments.
It's not that complicated though.

File: README.md [hunk 2]
```diff
@@ -31,26 +14,14 @@ The following algorithms will be implemented from scratch in Rust:
 - First-In-First-Out
```
Comment (NeoZ666) [inline README.md:14]: Shouldn't we add bnb, knapsack and others too here?

File: README.md [hunk 3]
```diff
@@ -31,26 +14,14 @@ The following algorithms will be implemented from scratch in Rust:
...
 - Single-Random-Draw
```
Comment (NeoZ666) [inline README.md:15]: Shouldn't we add bnb, knapsack and others too here?

File: src/algorithms/bnb.rs (hunk 1 above)
Comment (NeoZ666) [reply src/algorithms/bnb.rs:11]: I will come up with a better comment

File: README.md (hunk 3 above)
Comment (delcin-raj) [reply README.md:15]: those are already implemented, so should change the wording as well

File: src/algorithms/bnb.rs (hunk 1 above)
Comment (NeoZ666) [reply src/algorithms/bnb.rs:11]: /// Struct MatchParameters encapsulates target_for_match, match_range, and target_feerate.
/// Reduces the number of arguments in fn bnb, optimizing register usage and minimizing stack allocation.
/// Fits within 1 XMM register and 1 GPR for improved performance.
/// Link for reference : https://users.rust-lang.org/t/avoiding-too-many-arguments-passing-to-a-function/103581

File: README.md (hunk 3 above)
Comment (NeoZ666) [reply README.md:15]: ' Other algorithms implemented : '

**Comment by rajarshimaitra** [comment]
Looks amazing. I really love the new structure and docs. Don't have specific comments right now. I will check the rustdoc lsater when this is more finalized. 

I would like to see some more explainer text at `example.rs`, as thats the main source from which downstream devs will learn how to integrate.

Rest looks solid. Hats off too all those who took the lib so far!! This will be major contribution to rust-bitcoin ecosystem.

**Pull request merged**